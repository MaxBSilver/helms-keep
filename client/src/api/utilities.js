export const host = 'http://localhost:5000/';

export const mockTask = {
  id: 1234,
  isCompleted: false, 
  message: 'Clean the kitchen'
}

export const mockNote = {
  id: 1234,
  title: 'My First Quest',
  tasks: [mockTask, mockTask]
}