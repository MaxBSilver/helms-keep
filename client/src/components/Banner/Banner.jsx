import React, { Component } from "react";
import ExperienceBar from "../../containers/ExperienceBar/ExperienceBar";
import Rogue from "../../assets/Rogue.png";
import CentaurLady from "../../assets/centaurlady.png"
import Elf from "../../assets/elf_f_hit_anim_f0.png";
import Knight from "../../assets/knight_idle.png";
import { addLvl, addExp } from "../../actions";
import { connect } from "react-redux";

export class Banner extends Component {

  selectAvatar = () => {
    const { lvl } = this.props.user
    switch (true) {
      case (lvl > 4 && lvl < 10):
        return Knight;
      case (lvl > 10 && lvl < 15):
        return CentaurLady;
      default:
        return Elf;
    }
  };

  render() {
    const avatar = this.selectAvatar();
    return (
      <section className="Banner">
        <div className="info-cntr">
          <section>
            <span>{this.props.user.lvl}</span>
            <img src={avatar} style={{ width: 25 }} />
            <span>{this.props.user.name}</span>
          </section>
          <ExperienceBar user={this.props.user} />
        </div>
      </section>
    );
  }
}
export const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(Banner);
