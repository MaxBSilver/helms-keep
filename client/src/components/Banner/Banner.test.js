import React from 'react';
import { shallow } from 'enzyme';
import { Banner } from './Banner';
import { mockUser } from '../../api/utilities';

describe('Banner', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<Banner user={mockUser}/>);
    instance = wrapper.instance();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('selectAvatar', () => {
    it('should return elf as default', () => {
      const result = instance.selectAvatar();
      const expected = "elf_f_hit_anim_f0.png"
      expect(result).toEqual(expected);
    });

    it('should return knight if user level is between 5 and 9', () => {
      let lvl2User = Object.assign(mockUser, { lvl: 7 })
      wrapper = shallow(<Banner user={lvl2User} />);
      instance = wrapper.instance();
      const result = instance.selectAvatar();
      const expected = "knight_idle.png"
      expect(result).toEqual(expected);
    });

    it('should return centaur if user level is between 5 and 9', () => {
      let lvl2User = Object.assign(mockUser, { lvl: 12 })
      wrapper = shallow(<Banner user={lvl2User} />);
      instance = wrapper.instance();
      const result = instance.selectAvatar();
      const expected = "centaurlady.png"
      expect(result).toEqual(expected);
    });
  });


});