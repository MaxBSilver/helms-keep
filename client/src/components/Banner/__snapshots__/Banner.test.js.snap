// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should match snapshot 1`] = `
<section
  className="Banner"
>
  <div
    className="info-cntr"
  >
    <section>
      <span>
        1
      </span>
      <img
        src="elf_f_hit_anim_f0.png"
        style={
          Object {
            "width": 25,
          }
        }
      />
      <span>
        Jacob
      </span>
    </section>
    <ExperienceBar
      user={
        Object {
          "exp": 0,
          "lvl": 1,
          "name": "Jacob",
          "nextLvl": 100,
        }
      }
    />
  </div>
</section>
`;
